mdpow/tests/test_automated_dihedral_analysis.py

```diff
@@ -0,0 +1,215 @@
+        i = 0
+        while i < len(groups):
+            assert groups[i].all() == self.check_groups[i].all()
+            i+=1
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="scipy circvar gives wrong answers")
+    def test_dihedral_groups_ensemble(self, gen_data):
+
+        df = gen_data[2]
+
+        dh1_result = df.loc[df['selection'] == 'O1-C2-N3-S4']['dihedral']
+        dh1_mean = circmean(dh1_result, high=180, low=-180)
+        dh1_var = circvar(dh1_result, high=180, low=-180)
+
+        dh2_result = df.loc[df['selection'] == 'C13-C14-C15-C20']['dihedral']
+        dh2_mean = circmean(dh2_result, high=180, low=-180)
+        dh2_var = circvar(dh2_result, high=180, low=-180)
+
+        # default relative tolerance is 1e-6 for pytest.approx
+        dh1_mean == pytest.approx(self.DG_O1C2N3S4_mean)
+        dh1_var == pytest.approx(self.DG_O1C2N3S4_var)
+
+        dh2_mean == pytest.approx(self.DG_C13141520_mean)
+        dh2_var == pytest.approx(self.DG_C13141520_var)
+        
+    def test_save_df(self, gen_data, SM25_tmp_dir):
+        dihedrals.save_df(df=gen_data[2], df_save_dir=SM25_tmp_dir, molname='SM25')
+        assert SM25_tmp_dir / 'SM25' / 'SM25_full_df.bz2'
+                                        
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="scipy circvar gives wrong answers") 
+    def test_periodic_angle(self, gen_data):
+
+        df_aug = gen_data[3]
+
+        aug_dh2_result = df_aug.loc[df_aug['selection'] == 'C13-C14-C15-C20']['dihedral']
+
+        aug_dh2_mean = circmean(aug_dh2_result, high=180, low=-180)
+        aug_dh2_var = circvar(aug_dh2_result, high=180, low=-180)
+
+        aug_dh2_mean == pytest.approx(self.ADG_C13141520_mean)
+        aug_dh2_var == pytest.approx(self.ADG_C13141520_var)
+
+    def test_save_fig(self, SM25_tmp_dir):
+        dihedrals.automated_dihedral_analysis(dirname=SM25_tmp_dir, figdir=SM25_tmp_dir,
+                                              resname=self.resname, molname='SM25',
+                                              solvents=('water',))
+        assert SM25_tmp_dir / 'SM25' / 'SM25_C10-C5-S4-O11_violins.pdf'
+
+    def test_DataFrame_input(self, SM25_tmp_dir):
+        test_df = pd.DataFrame([['C1-C2-C3-C4', 'water', 'Coulomb', 0, 0, 60.0],
+                                ['C1-C2-C3-C5', 'water', 'Coulomb', 0, 0, 60.0]],
+                                [1,2],['selection', 'solvent', 'interaction', 'lambda', 'time', 'dihedral'])
+        dihedrals.automated_dihedral_analysis(dirname=SM25_tmp_dir, figdir=SM25_tmp_dir,
+                                              resname=self.resname,
+                                              solvents=('water',), dataframe=test_df)
+
+        with open('mdpow.log', 'r') as file:
+            log_content = file.read()
+            return_value = log_content.find('Proceeding with results DataFrame provided.')
+
+        assert return_value != -1
```

Comment:
What is tested here?

Looking in the log file is a very indirect way to check for things (and may fail when multiple tests write to it). Is there a more direct to test what you want to test? Doesn't `dihedrals.automated_dihedral_analysis` return something that we can work with?